packages/@lwc/babel-plugin-component/src/decorators/index.ts

```diff
@@ -142,29 +152,32 @@ function generateInvalidDecoratorError(path) {
     }
 }
 
-function collectDecoratorPaths(bodyItems) {
-    return bodyItems.reduce((acc, bodyItem) => {
+function collectDecoratorPaths(bodyItems: NodePath<types.Node>[]): NodePath<types.Decorator>[] {
+    return bodyItems.reduce((acc: NodePath<types.Decorator>[], bodyItem) => {
         const decorators = bodyItem.get('decorators');
-        if (decorators && decorators.length) {
-            acc.push(...decorators);
+        if (Array.isArray(decorators)) {
+            acc.push(...(decorators as NodePath<types.Decorator>[]));
         }
         return acc;
     }, []);
 }
 
-function getDecoratorMetadata(decoratorPath) {
-    const expressionPath = decoratorPath.get('expression');
+function getDecoratorMetadata(decoratorPath: NodePath<types.Decorator>): DecoratorMeta {
+    const expressionPath = decoratorPath.get('expression') as NodePath<types.Node>;
 
-    let name;
+    let name: 'api' | 'track' | 'wire';
     if (expressionPath.isIdentifier()) {
-        name = expressionPath.node.name;
+        name = expressionPath.node.name as 'api' | 'track' | 'wire';
```

Comment:
Type alias for `'api' | 'track' | 'wire'` would be nice.

Reply:
Say hello to `LwcDecoratorName`!